src/command_modules/azure-cli-backup/azure/cli/command_modules/backup/custom_wl.py

```diff
@@ -0,0 +1,275 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import azure.cli.command_modules.backup.custom_help as custom_help
+# pylint: disable=import-error
+
+from knack.log import get_logger
+
+from azure.mgmt.recoveryservicesbackup.models import AzureVMAppContainerProtectionContainer, ProtectionContainerResource
+
+from azure.cli.core.util import CLIError, sdk_no_wait
+from azure.cli.command_modules.backup._client_factory import (
+    backup_protected_items_cf, backup_protection_containers_cf, protectable_containers_cf)
+
+logger = get_logger(__name__)
+
+fabric_name = "Azure"
+default_policy_name = "DefaultPolicy"
+os_windows = 'Windows'
+os_linux = 'Linux'
+password_offset = 33
+password_length = 15
+
+# Mapping of workload type
+workload_type_map = {'MSSQL': 'SQLDataBase',
+                     'SAPHANA': 'SAPHanaDatabase'}
+
+
+def show_wl_policy(client, resource_group_name, vault_name, name):
+    return [client.get(vault_name, resource_group_name, name)]
+
+
+def list_wl_policies(client, resource_group_name, vault_name, workload_type, backup_management_type):
+    if workload_type is None:
+        raise CLIError(
+            """
+            Workload type is required for Azure Workload.
+            """)
+
+    if backup_management_type is None:
+        raise CLIError(
+            """
+            Backup Management Type needs to be specified for Azure Workload.
+            """)
+
+    workload_type = workload_type_map[workload_type]
+
+    filter_string = custom_help._get_filter_string({
+        'backupManagementType': backup_management_type,
+        'workloadType': workload_type})
+
+    policies = client.list(vault_name, resource_group_name, filter_string)
+    return custom_help._get_list_from_paged_response(policies)
+
+
+def list_protectable_containers(cmd, resource_group_name, vault_name, container_type="AzureWorkload"):
+    filter_string = custom_help._get_filter_string({
+        'backupManagementType': container_type
+        })
+
+    paged_containers = (protectable_containers_cf(cmd.cli_ctx)).list(vault_name, resource_group_name, fabric_name, filter_string)
+
+    return custom_help. _get_list_from_paged_response(paged_containers)
+
+
+def register_wl_container(cmd, client, vault_name, resource_group_name, workload_type, resource_id, container_type):
+    workload_type = workload_type_map[workload_type]
+
+    # Extracting friendly container name from resource id
+    container_name = resource_id.split('/')[-1]
+
+    containers = list_protectable_containers(cmd, resource_group_name, vault_name)
+
+    for container in containers:
+        if container.properties.friendly_name == container_name:
```

Comment:
check for the native name as 2 containers can have the same friendly name

Reply:
Need to discuss further.